lib/config/expression.cpp

```diff
@@ -675,20 +675,11 @@ ExpressionResult SetConstExpression::DoEvaluate(ScriptFrame& frame, DebugHint *d
 {
 	auto globals = ScriptGlobal::GetGlobals();
 
-	auto attr = globals->GetAttribute(m_Name);
-
-	if (dynamic_pointer_cast<ConstEmbeddedNamespaceValue>(attr)) {
-		std::ostringstream msgbuf;
-		msgbuf << "Value for constant '" << m_Name << "' was modified. This behaviour is deprecated.\n";
-		ShowCodeLocation(msgbuf, GetDebugInfo(), false);
-		Log(LogWarning, msgbuf.str());
-	}
-
```

Comment:
`SetAttribute()` did not do checks that `Set()` does. So the deprecated behavior is no longer allowed but raises an error.

```
Icinga 2 (version: v2.13.0-547-g48eb1aa69)
Type $help to view available commands.
<1> => const x = 23
null
<2> => const x = 42
Constant must not be modified.
```

Reply:
Yes, I also saw the code causing this in the changeset. But what's the reason?

Reply:
Because continuing to support the deprecated behavior would require implementing something to ignore const flags in namespaces. I don't think that doing so is worth it just to keep behavior [deprecated since v2.10.0](1a8692d972c4d9c5e6aa79a042cedd1b84941b8a).